sherlock/db/migrations/000062_environment_v3.up.sql

```diff
@@ -0,0 +1,54 @@
+alter table environments
+    add constraint name_valid
+        check (name != ''  and name similar to '[a-z0-9]([-a-z0-9]*[a-z0-9])?');
+
+alter table environments
+    add constraint owner_id_present
+        check (owner_id is not null or
+               (legacy_owner is not null and legacy_owner != ''));
+
+alter table environments
+    add constraint lifecycle_valid
+        check ((lifecycle = 'template' and template_environment_id is null) or
+               (lifecycle = 'dynamic' and template_environment_id is not null) or
+               (lifecycle = 'static' and
+                base != '' and
+                default_cluster_id is not null and
+                requires_suitability is not null));
```

Comment:
I'm not sure this is quite right -- https://github.com/broadinstitute/sherlock/blob/main/sherlock/internal/deprecated_models/v2models/environment.go#L189 has all of the static checks also apply to dynamic. Template gets to break some rules because we won't actually end up creating it in our infrastructure, but both dynamic and static need to have base/default cluster/requires suitability